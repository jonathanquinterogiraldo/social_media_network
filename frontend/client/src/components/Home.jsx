import React, { useEffect, useState } from "react"
import axios from 'axios'
import { Form, Button, Col, Container, Row } from 'react-bootstrap'
import { Link } from 'react-router-dom'
import { useForm } from 'react-hook-form'
import { useHistory } from 'react-router-dom'
import swal from 'sweetalert'
import Header from "./Header"

const apiUrl = process.env.REACT_APP_API_URL

function Home() {

  const history = useHistory()

  const { handleSubmit } = useForm();

  const [posts, setPosts] = useState([])

  useEffect(() => {
    axios({
      method: 'GET',
      baseURL: apiUrl,
      url: '/posts/posts',
      headers: {
        Authorization: `Bearer ${localStorage.getItem('token')}`
      }
    }).then(
      ({ data }) => {
        if (data.allPosts) {
          setPosts(data.allPosts)
        }
      }
    ).catch(
      error => {
        console.log(error)
      }
    )
  }, [])

  // const onSubmit = () => {
  //   swal({
  //     text: '¿Desea cerrar la sesión?',
  //     icon: 'warning',
  //     buttons: ['No', 'Sí']
  //   }).then(response => {
  //     if (response) {
  //       localStorage.removeItem('token')
  //       history.push('/login')
  //     }
  //   })
  // }

  const onSubmitPost = () => {
    history.push('/post')
  }

  return (
    <div>
      <Header />
      {posts.map((post) => <p key={post._id}>{post.content}</p>)}
      {/* <Form onSubmit={handleSubmit(onSubmit)}>
        <Button variant="danger" type='submit'>Salir</Button>
      </Form> */}


      <Button variant="success" onClick={onSubmitPost}>Nuevo Post</Button>


    </div>
  )
}

export default Home
