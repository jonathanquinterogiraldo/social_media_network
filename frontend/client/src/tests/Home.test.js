import React from 'react'
import '@testing-library/jest-dom'
import { shallow } from 'enzyme'
import Home from '../components/Home'
import { expect } from '@jest/globals'

describe('Unit Test <Home /> Component', () => {
    test('should mount <Home /> correctly', () => {

        const wrapper = shallow(<Home />)

        expect(wrapper).toMatchSnapshot() 
        
    })    
})