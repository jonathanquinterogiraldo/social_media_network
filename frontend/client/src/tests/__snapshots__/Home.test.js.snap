// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Unit Test <Home /> Component should mount <Home /> correctly 1`] = `
<div
  className="Home"
>
  <Form
    inline={false}
    onSubmit={[Function]}
  >
    <Button
      active={false}
      disabled={false}
      type="submit"
      variant="danger"
    >
      Salir
    </Button>
  </Form>
</div>
`;
